Test/Utils/ContrailAPI/VirtualNetwork.ps1

```diff
@@ -1,118 +1,142 @@
 function Remove-ContrailVirtualNetwork {
-    Param ([Parameter(Mandatory = $true)] [string] $ContrailUrl,
-           [Parameter(Mandatory = $true)] [string] $AuthToken,
-           [Parameter(Mandatory = $true)] [string] $NetworkUuid,
+    Param ([Parameter(Mandatory = $true)] [ContrailNetworkManager] $API,
+           [Parameter(Mandatory = $true)] [string] $Uuid,
            [bool] $Force = $false)
 
-    $NetworkUrl = $ContrailUrl + "/virtual-network/" + $NetworkUuid
-    $Network = Invoke-RestMethod -Method Get -Uri $NetworkUrl -Headers @{"X-Auth-Token" = $AuthToken}
+    $Network = $API.Get('virtual-network', $Uuid, $null)
+
+    $Props = $Network.'virtual-network'.PSobject.Properties.Name
+
+    $VirtualMachines = $null
+    $IpInstances = $null
 
-    if ($Force) {
-        # TODO remove this outer if, this is only a quick workaround,
-        # because sometimes the fields below are empty,
-        # and that's a failure in strict mode (I guess).
+    if($Props -contains 'virtual_machine_interface_back_refs') {
         $VirtualMachines = $Network.'virtual-network'.virtual_machine_interface_back_refs
+    }
+
+    if($Props -contains 'instance_ip_back_refs') {
         $IpInstances = $Network.'virtual-network'.instance_ip_back_refs
+    }
 
-        if ($VirtualMachines -or $IpInstances) {
-            if (!$Force) {
-                Write-Error "Couldn't remove network. Resources are still referred. Use force mode"
-                return
-            }
+    if ($VirtualMachines -or $IpInstances) {
+        if (!$Force) {
+            Write-Error "Couldn't remove network. Resources are still referred. Use force mode"
+            return
+        }
 
-            # First we have to remove resources referred by network instance in correct order:
-            #   - Instance IPs
-            #   - Virtual machines
+        # First we have to remove resources referred by network instance in correct order:
+        #   - Instance IPs
+        #   - Virtual machines
+        if($IpInstances) {
             ForEach ($IpInstance in $IpInstances) {
-                Invoke-RestMethod -Uri $IpInstance.href -Headers @{"X-Auth-Token" = $AuthToken} -Method Delete | Out-Null
+                $API.Delete('instance-ip', $IpInstance.'uuid', $null)
             }
+        }
 
+        if($VirtualMachines) {
             ForEach ($VirtualMachine in $VirtualMachines) {
-                Invoke-RestMethod -Uri $VirtualMachine.href -Headers @{"X-Auth-Token" = $AuthToken} -Method Delete | Out-Null
+                $API.Delete('virtual-machine-interface', $VirtualMachine.'uuid', $null)
             }
         }
     }
 
-    # We can now remove the network
-    Invoke-RestMethod -Uri $NetworkUrl -Headers @{"X-Auth-Token" = $AuthToken} -Method Delete | Out-Null
+    $API.Delete('virtual-network', $Uuid, $null)
+}
+
+# TODO support multiple subnets per network
+# TODO return a class (perhaps use the class from MultiTenancy test?)
+function Add-OrReplaceNetwork {
```

Comment:
This function and all other Add-OrReplace* will be changed into parameter for New- functions